server/testrun/testrun.go

```diff
@@ -347,42 +353,61 @@ finishedloop:
 
 }
 
-func cleanTestData(dirtyData map[string]string) map[string]map[string]map[string]float32 {
+// cleanTestData cleans up the testing data that comes back from the agents.
+// The test data that comes back raw from the agents is "dirty", meaning it is designed to be as flexible
+// as possible.
+func cleanTestData(dirtyData map[string]string) (map[string]map[string]map[string]interface{}, error) {
```

Comment:
It seems like this function exists primarily to convert the string escaped metrics back into JSON. I think much of this could be avoided by making a couple changes in `agent/tasks/executetestrun.go`.

On line 43, change
``` go
gatheredData := map[string]string{}
```
to 
``` go
gatheredData := map[string]json.RawMessage{}
```

On line 118, change
``` go
gatheredData[thisTarget] = string(cmdOutput.Bytes())
```
to
``` go
gatheredData[thisTarget] = cmdOutput.Bytes()
```

This will prevent the string from being escaped in the first place.

At which point this function could be simplified to:
``` go
	retMap := make(map[string]map[string]map[string]interface{})
	for sourceUUID, agentData := range dirtyData {
		var agentMap map[string]map[string]interface{}
		err := json.Unmarshal([]byte(agentData), &agentMap)
		if err != nil {
			return nil, errors.New("Failed to unmarshal raw test data")
		}
		retMap[sourceUUID] = agentMap
	}
```